test/unit/v1.0.0/normalize-v1.test.js

```diff
@@ -63,15 +63,215 @@ describe('v1.0.0 normalization', function () {
                     retainIds: true
                 };
 
-            transformer.normalize(fixture.raw, options, function (err, converted) {
+            transformer.normalize(fixture.raw, options, function (err, normalized) {
                 expect(err).to.not.be.ok;
 
                 // remove `undefined` properties for testing
-                converted = JSON.parse(JSON.stringify(converted));
+                normalized = JSON.parse(JSON.stringify(normalized));
 
-                expect(converted).to.eql(fixture.normalized);
+                expect(normalized).to.eql(fixture.normalized);
                 done();
             });
         });
     });
+
+    describe('special cases', function () {
+        describe('auth', function () {
+            it('should override auth with legacy properties if both are present', function (done) {
+                var options = {
+                        normalizeVersion: '1.0.0',
+                        retainIds: true
+                    },
+                    source = {
+                        id: 'bd79f978-d862-49f1-9cea-7c71a762cc12',
+                        currentHelper: 'basicAuth',
+                        helperAttributes: {
+                            id: 'basic',
+                            username: 'username',
+                            password: 'password'
+                        },
+                        auth: {
+                            type: 'noauth'
```

Comment:
This might not be an ideal test case. We need to assert for a valid auth in legacy and non legacy format. `noauth` goes through an early bail out code flow.